app/models/application.rb

```diff
@@ -132,6 +132,8 @@ def savings_investment_valid?
   def benefits=(val)
     super
     self.application_type = benefits? ? 'benefit' : 'income'
+    self.dependents = nil if benefits?
+    self.application_outcome = outcome_from_dwp_result if benefits? && last_benefit_check.present?
```

Comment:
I'm thinking that it would be cleaner to have a method to convert the `Application` from `r1` to `r2`. Doing these assignments in setters is not obvious and I've ran into a similar problem in `children=` setter, which has a conditional in. 

I think setters should be setters and if there are conditions required for them, we should rather use validations or different methods. 

What does @dotemacs think?